go-controller/pkg/node/gateway.go

```diff
@@ -373,14 +373,33 @@ type bridgeConfiguration struct {
 }
 
 // updateInterfaceIPAddresses sets and returns the bridge's current ips
-func (b *bridgeConfiguration) updateInterfaceIPAddresses() ([]*net.IPNet, error) {
+func (b *bridgeConfiguration) updateInterfaceIPAddresses(node *kapi.Node) ([]*net.IPNet, error) {
 	b.Lock()
 	defer b.Unlock()
 	ifAddrs, err := getNetworkInterfaceIPAddresses(b.bridgeName)
-	if err == nil {
-		b.ips = ifAddrs
+	if err != nil {
+		return nil, err
+	}
+
+	// For DPU, here we need to use the DPU host's IP address which is the tenant cluster's
+	// host internal IP address instead of the DPU's external bridge IP address.
+	if config.OvnKubeNode.Mode == types.NodeModeDPU {
+		nodeAddrStr, err := util.GetNodePrimaryIP(node)
+		if err != nil {
+			return nil, err
+		}
+		nodeAddr := net.ParseIP(nodeAddrStr)
+		if nodeAddr == nil {
+			return nil, fmt.Errorf("failed to parse node IP address. %v", nodeAddrStr)
+		}
+		ifAddrs, err = getDPUHostPrimaryIPAddresses(nodeAddr, ifAddrs)
+		if err != nil {
+			return nil, err
+		}
 	}
-	return ifAddrs, err
+
+	b.ips = ifAddrs
```

Comment:
Possibly need to move this before `config.OvnKubeNode.Mode == types.NodeModeDPU` check.

Reply:
@zshi-redhat

Reply:
After following the code for bridgeForInterface() in gateway.go, b.ips should be the ip address of the DPU-Host.